src/config.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Spark History Server configuration."""
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from constants import (
+    CONFIG_KEY_S3_ACCESS_KEY,
+    CONFIG_KEY_S3_CREDS_PROVIDER,
+    CONFIG_KEY_S3_ENDPOINT,
+    CONFIG_KEY_S3_LOGS_DIR,
+    CONFIG_KEY_S3_SECRET_KEY,
+)
+from utils import WithLogging
+
+
+class AbstractSparkHistoryServerConfig(WithLogging, ABC):
```

Comment:
**question** are we using Abstract because we imagine multiple implementations of this class? Which one?